### DISCUSSION TITLE
基于 Ant Design Web3 的前端 DApp 研发教程

### DISCUSSION THREAD

**Discussion by yutingzhao1991** [ant-design/ant-design-web3#178] [Ideas]
发现目前懂区块链、智能合约相关概念的前端开发人员还是比较少，可以搞一个前端 DApp 研发教程，也可以吸引更多人关注 Ant Design Web3。

大家可以看看有没有什么想法和建议。

**Comment by yutingzhao1991** [comment]
1. 北大肖臻的区块链公开课：https://www.bilibili.com/video/BV1Vt411X7JF/?vd_source=9461a701354aa450cddb02f87ed560a4
2. WTF Solidity 和 Ethersjs 的教程：https://wtf.academy/
3. https://myfirstnft.info/

这些也可以参考一下

**Comment by kiner-tang** [comment]
是指出一个文档教程吗？

**Reply by yutingzhao1991** [reply]
嗯，搞一个教程，教大家怎么开发一个 dapp

**Comment by yutingzhao1991** [comment]
我大概想了一个大纲：

- 课程介绍，介绍课程大概内容，面向对象，前置要求，如何参与贡献等
- DApp 是什么？
   - DApp 概要
   - 区块链和智能合约
   - DApp 如何调用智能合约
- 开发 DApp 实战
   - 初始化项目
   - 连接区块链
   - 调用合约
- 开发智能合约
   - 初始化合约
   - 合约开发和测试
   - 编译和部署合约
- 扩展知识
   - 什么是 L2
   - 等等

**Comment by Likang0122** [comment]
![image](https://github.com/ant-design/ant-design-web3/assets/21904785/61798809-7749-4438-b69f-82640453fd9b)
我有过一篇类似的文档，可以在此基础上再扩充下

**Reply by yutingzhao1991** [reply]
写在哪里的？可以看看吗

**Reply by kiner-tang** [reply]
@Likang0122 可以把链接发出来看看么

**Reply by yutingzhao1991** [reply]
私下问了下，这个是 @Likang0122 的一个没补分享文档，没有公开连接，后面他可以一起来参与教程编写

**Comment by dxiongya** [comment]
感觉可以参考郭宇那篇文章，然后规划下课程内容：
1. 介绍DAPP涉及的一些相关技术上下文（智能合约，Provider（InjectProvider, RpcProvider），区块链节点(RPC服务商，自建)，服务端）
2. 技术工具（react，vue中常用的一些工具，包括antd web3）
3. 围绕Provider，Account，Contract构建讲解一下
4. 课程再给一个实践（1. 合约构建（看是否深入，不深入即remix，深入就常见的那一套走一遍）2. 如何使用antd web3快速启动一个项目+连接钱包+合约调用+交易状态维护）
5. 扩展...

**Reply by yutingzhao1991** [reply]
你说的是  https://guoyu.mirror.xyz/RD-xkpoxasAU7x5MIJmiCX4gll3Cs0pAd5iM258S1Ek 这篇吧

**Reply by dxiongya** [reply]
嗯，是的，只不过课程的内容会有所不同，类似上面，我个人的想法～

**Reply by yutingzhao1991** [reply]
嗯，我们的内容应该会更篇入门，更适合新手。

合约那部分已经有一个版本了，基于 remix。总体思路就是以一个简单的 dapp 开发实践入门，过程中会介绍相关概念。

**Comment by restartgr** [comment]
目前现状：

1. 对于前端来说antd的名声在国内react基本上传播很广，但是前端开始学习web3的很少
2. web3合约开发之前可能接触过前端页面重构的不多，了解antd的不多

所以重要的是如何吸引两波人参与进来，我个人通过学习和交流的想法是

1. 如果有时间可以以antdweb3名义组织一些comp，内容可以是solidity入门，或者dapp教程，其中直接引用antdweb3组件库开发，通过对比数据，可以展现出通过组件库开发高效的效果，同时通过免费comp，能增加受众，目前参加的solidity comp和move comp基本都能达到500+人数，感觉受众还是不错的。
2. 是否能赞助一些hackathon提高知名度。

单独开展dapp教程固然好，但是只靠蚂蚁链或antd的名气可能不太好将两波人融合出去，像之前的wtf也是一阵子后才爆火，只是个人的意见

**Reply by yutingzhao1991** [reply]
是的，你的想法和我们一样。我们做这个课程本身也是切入这两种人群的场景。搞前端的入门区块链，搞区块链的要搭一个 dapp。solidity 的教程 wtf 做得已经不错了，我们会专注 dapp 开发入门这块。解决 dapp 前端开发的痛点，帮助开发者从 web2 到 web3。

至于如何吸引两波人参与进来，一方面课程做好，简单容易上手，满足上面说的两波人的需求。

另外你说的比赛和 hackathon 这些也会考虑。其实之前 Hangzhou ETH Hackathone 我们去了。明年应该也还会去（有项目方会支持哈哈 https://zan.top/home 这个可以了解下~）。

**Reply by restartgr** [reply]
我觉得目前项目的优势是
1. antd的珠玉在前，有大厂光环。
2. 目前web3合约开发、dapp搭建确实需要一个组件库，目前市场是空缺的。

但是因为蚂蚁链是联盟链，不知道能不能收到公链开发们的认可，愿意去使用。
我今天从推上看到这个新项目感觉还是很有搞头的，不过我对于web3只能算刚学习的初学者，希望能以后更多的参与共建吧。

**Reply by yutingzhao1991** [reply]
<img width="330" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/1061968/f1e5605c-596a-4d32-92fb-b9771f14e615">

嗯，感谢支持，欢迎共建。也可以加群，不过我们项目内容都会直接在 github 上讨论，但是群里面有时候也会聊一些 web3 相关内容。

**Reply by restartgr** [reply]
嗯感谢邀请，早些时间拜托大圣老师已经拉进群了，希望跟大佬们多学习

**Reply by yutingzhao1991** [reply]
另外：
- 蚂蚁链也有在做公链相关内容，伴随未来监管越来越完善，公链也会在国内至少在香港得到支持。
- web3 组件我们之前调研过，其实也有一些了，尤其是连接钱包有很多商业化的公司会做得更好。不过一个纯粹的不仅仅是连接钱包的 UI 组件库，还可以支持不同的链，我相信 Ant Design Web3 能够做好。

**Comment by yutingzhao1991** [comment]
在写 『开发 DApp 实战』 这部分的时候发现我们需要在『DApp 是什么？』这部分让用户先把钱包搞定，能够玩起来。

目前的想法就是我们把课程实战后完成的 Demo 部署起来作为一个 DApp 的样例给开发者体验，这样整个就串起来了。

**Reply by yutingzhao1991** [reply]
@jeasonstudio

**Comment by mlxglee** [comment]
是否能借助最近 BRC20 的热度，进行一下宣传

**Reply by yutingzhao1991** [reply]
我们还不支持比特币，最近准备支持 solana，solana 最近也比较火。

**Comment by yutingzhao1991** [comment]
https://github.com/WTFAcademy/WTF-Dapp/pull/1

会和 WTF 的课程做一些整合。目前我的想法是 WTF-Dapp 中的内容会专注 Dapp 开发，也会更完善和详细。

Ant Design Web3 中的内容则会包括区块链的简介和合约的开发引导，但是整体上会更简单。

其中部分内容会重复，但是目前感觉不需要做自动化同步，各自维护就行，未来也不需要保持一致，两边互相补充引导即可。

**Reply by yutingzhao1991** [reply]
@jeasonstudio @Likang0122
